oreClient/src/main/assets/pages/project/NewProject.vue

```diff
@@ -0,0 +1,144 @@
+                        </div>
+
+                        <div class="form-group">
+                            <label for="projectCategory">Owner</label>
+                            <select id="projectCategory" name="owner" class="form-control" v-model="owner" required>
+                                <option :value="currentUser.name">{{ currentUser.name }}</option>
+                                <option v-for="membership in availableOwners" :value="membership.organization.name">{{ membership.organization.name }}</option>
+                            </select>
+                        </div>
+
+                        <button @click="create()" class="btn btn-primary">Create project</button>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+    import {mapState} from "vuex";
+    import {Category, Permission} from "../../enums";
+    import {API} from "../../api"
+
+    export default {
+        data() {
+            return {
+                projectName: "",
+                pluginId: "",
+                projectDescription: "",
+                category: null,
+                owner: "",
+                availableOwners: []
+            };
+        },
+        computed: {
+            categories() {
+                return Category
+            },
+            ...mapState('global', [
+                'currentUser',
+                'memberships'
+            ])
+        },
+        methods: {
+            create() {
+                let error = false;
+                const messages = [];
+
+                if(!this.projectName || this.projectName === "") {
+                    messages.push("Project name is required");
+                    error = true;
+                }
+                if(!this.pluginId || this.pluginId === "") {
+                    messages.push("Plugin id is required");
+                    error = true;
+                }
+                if(!this.category) {
+                    messages.push("Category is required");
+                    error = true;
+                }
+                if(!this.owner || this.owner === "") {
+                    messages.push("Owner is required");
+                    error = true;
+                }
+
+                if(error) {
+                    this.$store.commit({
+                        type: 'addAlerts',
+                        level: 'error',
+                        messages: messages
+                    });
+                } else {
+                    API.request("projects", "POST", {
+                        name: this.projectName,
+                        plugin_id: this.pluginId,
+                        category: this.category,
+                        description: this.projectDescription,
+                        owner_name: this.owner
+                    }).then((data) => {
+                        this.$router.push({path: `/${data.namespace.owner}/${data.namespace.slug}`})
```

Comment:
Why not do name based routing, and also pass it extra info we have about the project. We have the whole fetched project, so might as well just pass it in. Lookup by namespace is also a bit slower currently.